Fix DST handling in TOD (#29042)
home-assistant/home-assistant.github.io: source/_integrations/tod.markdown

```

In the above example, the next day `sunrise` is calculated as a time range end.

## Daylight Savings Time Handling

> The ToD sensor handles the following cases where the sensor interval

Reviewer: ```suggestion
The ToD sensor handles the following cases where the sensor interval:
```

- does not exist at all
- stops at a non-existent time
- starts at a non-existent time.

To help understand all 3 cases, actual examples are provided below.

### Case 1: Sensor Interval Does Not Exist

Let's make the following assumptions:
- Daylight Savings starts at 2am
- On the DST day, the ToD sensor interval is from non-existent 2:30am to non-existent 2:40am.

> In this case, the ToD sensor will not trigger at all since the 2:30am-2:40am interval does not exist on the day where time jumps from 2am to 3am.  However, on the following day the sensor resumes operating normally.

Reviewer: ```suggestion
In this case, the ToD sensor will not trigger since the 2:30am-2:40am interval does not exist on the day when time jumps from 2am to 3am. However, on the following day, the sensor resumed operating normally.
```

### Case 2: Sensor End Time Does Not Exist

> - Daylight Savings starts at 2am

Reviewer: ```suggestion

- Daylight Savings starts at 2am
```

- On the DST day, the ToD sensor interval is from 1:50am to non-existent 2:10am.

In this case, the ToD sensor will last 10 minutes starting at 1:50am and stop at 3am (the 2am-3am time is jumped over and does not exist).

> ### Case 3: Sensor Start Time Does Not Exist

Reviewer: ```suggestion
### Case 3: Sensor Start Time Does Not Exist

```

- Daylight Savings starts at 2am
- On the DST day, the ToD sensor interval is from non-existent 2:50am to 3:10am.

> In this case, the ToD sensor will last 10 minutes starting at 3:00am and stop at 3:10am (the 2am-3am time is jumped over and does not exist).

Reviewer: ```suggestion
In this case, the ToD sensor will last 10 minutes, starting at 3:00am and stopping at 3:10am (the 2am-3am time is jumped over and does not exist).
```
